components/world_news_api/sources/news-event/news-event.mjs

```diff
@@ -0,0 +1,165 @@
+import app from "../../world_news_api.app.mjs";
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+
+export default {
+  name: "New News",
+  description: "Emit new event when a new news is fetched. Calling this endpoint requires 1 point per page, up to 1000 news.",
+  key: "world_news_api-news-event",
+  version: "0.0.1",
+  type: "source",
+  props: {
+    app,
+    timer: {
+      label: "Polling Interval",
+      description: "Pipedream will poll the Docusign API on this schedule",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: DEFAULT_POLLING_SOURCE_TIMER_INTERVAL,
+      },
+    },
+    db: "$.service.db",
+    text: {
+      propDefinition: [
+        app,
+        "text",
+      ],
+    },
+    sourceCountries: {
+      propDefinition: [
+        app,
+        "sourceCountries",
+      ],
+    },
+    language: {
+      propDefinition: [
+        app,
+        "language",
+      ],
+    },
+    minSentiment: {
+      propDefinition: [
+        app,
+        "minSentiment",
+      ],
+    },
+    maxSentiment: {
+      propDefinition: [
+        app,
+        "maxSentiment",
+      ],
+    },
+    earliestPublishedDate: {
+      propDefinition: [
+        app,
+        "earliestPublishedDate",
+      ],
+    },
+    latestPublishedDate: {
+      propDefinition: [
+        app,
+        "latestPublishedDate",
+      ],
+    },
+    newsSources: {
+      propDefinition: [
+        app,
+        "newsSources",
+      ],
+    },
+    authors: {
+      propDefinition: [
+        app,
+        "authors",
+      ],
+    },
+    entities: {
+      propDefinition: [
+        app,
+        "entities",
+      ],
+    },
+    locationFilter: {
+      propDefinition: [
+        app,
+        "locationFilter",
+      ],
+    },
+  },
+  methods: {
+    emit(data) {
+      this.$emit(data, {
+        id: data.id,
+        summary: data.title,
+        ts: Date.now(),
+      });
+    },
+    setLastId(id) {
+      this.db.set("lastId", id);
+    },
+    getLastId() {
+      return this.db.get("lastId");
+    },
+  },
+  async run({ $ }) {
+    const ITEMS_PER_PAGE = 100;
+    const MAX_CALLS_PER_EXECUTION = 10;
+    const MAX_OFFSET = MAX_CALLS_PER_EXECUTION * ITEMS_PER_PAGE;
+    let offset = 0;
+
+    let available = 0;
+    let currentCallLastId;
+    const beforeCallLastId = this.getLastId();
+    const newsArr = [];
+    loop1:
+    do {
+      const params = {
```

Comment:
I suggest you use `lodash` for picking the configured props. It will be a lot cleaner and make the code easier to read. [Here's an example.](https://github.com/PipedreamHQ/pipedream/blob/master/components/gitlab/actions/create-epic/create-epic.mjs#L102)

Reply:
I took a slightly different path from your suggestion. There are some peculiarities in the way that some empty props works and I am not sure if `lodash.pick` would be a good idea, but hopefully, the changes will make the code more readable

Reply:
All optional props that are not defined by the user will evaluate to `undefined`.
Using `lodash.pick` and `lodash.pickBy`, it will filter out those props.